src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -110,14 +112,33 @@ void Run (DirectoryAssemblyResolver res)
 
 			// Put every assembly we'll need in the resolver
 			foreach (var assembly in ResolvedAssemblies) {
+				if (bool.TryParse (assembly.GetMetadata (AndroidSkipJavaStubGeneration), out bool value) && value) {
+					Log.LogDebugMessage ($"Skipping Java Stub Generation for {assembly.ItemSpec}");
+					continue;
+				}
 				res.Load (assembly.ItemSpec);
 			}
 
 			// However we only want to look for JLO types in user code
-			var assemblies = ResolvedUserAssemblies.Select (p => p.ItemSpec).ToList ();
-			var fxAdditions = MonoAndroidHelper.GetFrameworkAssembliesToTreatAsUserAssemblies (ResolvedAssemblies)
-				.Where (a => assemblies.All (x => Path.GetFileName (x) != Path.GetFileName (a)));
-			assemblies = assemblies.Concat (fxAdditions).ToList ();
+			List<string> assemblies = new List<string> ();
+			foreach (var asm in ResolvedUserAssemblies) {
+				if (bool.TryParse (assembly.GetMetadata (AndroidSkipJavaStubGeneration), out bool value) && value) {
```

Comment:
This results in a [CS0103](https://jenkins.mono-project.com/job/xamarin-android-pr-pipeline-release/637/console):

```
10:14:03    Tasks/GenerateJavaStubs.cs(125,24): error CS0103: The name 'assembly' does not exist in the current context [/Users/builder/jenkins/workspace/xamarin-android-pr-pipeline-release/xamarin-android/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj]
```

Perhaps you need to rename `asm` to `assembly` on line 124?